### PR TITLE
Simplify receipts root calculation

### PR DESCRIPTION
**Pull request by i1i1**

### CODE REVIEW

File: crates/pevm/src/chain/optimism.rs [hunk 1]
```diff
@@ -189,45 +187,34 @@ impl PevmChain for PevmOptimism {
...
+                            deposit_nonce: (spec_id >= SpecId::CANYON).then_some(account.nonce - 1),
+                            deposit_receipt_version: (spec_id >= SpecId::CANYON).then_some(1),
+                        };
+                        OpReceiptEnvelope::Deposit(receipt.with_bloom())
+                    }
+                }
             })
-            .collect();
+            .enumerate()
+            .map(|(index, receipt)| (alloy_rlp::encode_fixed_size(&index), receipt.encoded_2718()))
```
Comment (kien-rise) [inline crates/pevm/src/chain/optimism.rs:215]: Just curious, how does `.encoded_2718` accept zero arguments? 

I thought it always accept one argument? `receipt.encode_2718(&mut value_buffer);`

**Review by kien-rise — APPROVED** [review]
No blocking issues. But remember to run some tests, especially for optimism side.

File: crates/pevm/src/chain/optimism.rs (hunk 1 above)
Comment (i1i1) [reply crates/pevm/src/chain/optimism.rs:215]: Yes! You right, but this method is `.encoded_2718` :)

It is basically:
```rust
fn encoded_2718(&self) -> Vec<u8> {
    let mut buf = Vec::with_capacity(self.encode_2718_len());
    self.encode_2718(&mut buf);
    buf
}
```

File: crates/pevm/src/chain/ethereum.rs [hunk 2]
```diff
@@ -196,19 +193,11 @@ impl PevmChain for PevmEthereum {
                     TxType::Eip4844 => ReceiptEnvelope::Eip4844(receipt),
                     TxType::Eip7702 => ReceiptEnvelope::Eip7702(receipt),
                 }
```
[the remark this answers is not in the document]
Comment (i1i1) [reply crates/pevm/src/chain/ethereum.rs:195]: It doesn't actually implement any kind of `From` which is sad:
https://reth.rs/docs/reth/rpc/types/enum.ReceiptEnvelope.html#trait-implementations

File: crates/pevm/src/chain/optimism.rs (hunk 1 above)
Comment (kien-rise) [reply crates/pevm/src/chain/optimism.rs:215]: oh, cool. I didn't notice the difference between `encoded_2718` and  `encode_2718` (the letter `d`)

Comment (i1i1) [reply crates/pevm/src/chain/optimism.rs:215]: Also, it should be faster (but it's not worth it here), as we will allocate precisely once (`Vec::with_capacity`) and it won't reallocate

**Review by hai-rise — APPROVED** [review]
Nice!

**Pull request merged**